transmart-oauth/src/main/groovy/org/transmart/oauth/TransmartOauthGrailsPlugin.groovy

```diff
@@ -0,0 +1,174 @@
+package transmart.oauth
+
+import grails.plugin.springsecurity.SpringSecurityUtils
+import grails.plugins.*
+import org.springframework.beans.factory.config.MapFactoryBean
+import org.springframework.security.core.session.SessionRegistryImpl
+import org.springframework.security.ldap.authentication.ad.ActiveDirectoryLdapAuthenticationProvider
+import org.springframework.security.web.DefaultRedirectStrategy
+import org.springframework.security.web.access.AccessDeniedHandlerImpl
+import org.springframework.security.web.authentication.SimpleUrlAuthenticationFailureHandler
+import org.springframework.transaction.interceptor.TransactionInterceptor
+import org.springframework.web.util.IntrospectorCleanupListener
+import org.transmart.oauth.ActiveDirectoryLdapAuthenticationExtension
+import org.transmart.oauth.AuthSuccessEventListener
+import org.transmart.oauth.BadCredentialsEventListener
+import org.transmart.oauth.CurrentUserBeanFactoryBean
+import org.transmart.oauth.CurrentUserBeanProxyFactory
+import org.transmart.oauth.LdapAuthUserDetailsMapper
+import org.transmart.oauth.authentication.AuthUserDetailsService
+import org.transmart.oauth.authentication.BruteForceLoginLockService
+import org.transmartproject.core.users.User
+
+class TransmartOauthGrailsPlugin extends Plugin {
+
+    // the version or versions of Grails the plugin is designed for
+    def grailsVersion = "3.2.3 > *"
+    // resources that are excluded from plugin packaging
+    def pluginExcludes = [
+        "grails-app/views/error.gsp"
+    ]
+
+    // TODO Fill in these fields
+    def title = "Transmart Oauth" // Headline display name of the plugin
+    def author = "Your name"
+    def authorEmail = ""
+    def description = '''\
+Brief summary/description of the plugin.
+'''
+    def profiles = ['web']
+
+    // URL to the plugin's documentation
+    def documentation = "http://grails.org/plugin/transmart-oauth"
+
+    // Extra (optional) plugin metadata
+
+    // License: one of 'APACHE', 'GPL2', 'GPL3'
+//    def license = "APACHE"
+
+    // Details of company behind the plugin (if there is one)
+//    def organization = [ name: "My Company", url: "http://www.my-company.com/" ]
+
+    // Any additional developers beyond the author specified above.
+//    def developers = [ [ name: "Joe Bloggs", email: "joe@bloggs.net" ]]
+
+    // Location of the plugin's issue tracker.
+//    def issueManagement = [ system: "JIRA", url: "http://jira.grails.org/browse/GPMYPLUGIN" ]
+
+    // Online location of the plugin's browseable source code.
+//    def scm = [ url: "http://svn.codehaus.org/grails-plugins/" ]
+
+    Closure doWithSpring() {
+        { ->
+            xmlns context: "http://www.springframework.org/schema/context"
+            xmlns aop: "http://www.springframework.org/schema/aop"
+
+            sessionRegistry(SessionRegistryImpl)
```

Comment:
TBD: Do we need to configure a concurrent sessions limit like transmartApp does?